src/components/DeckMechanisms/index.js

```diff
@@ -139,18 +138,16 @@ export default class DeckMechanisms extends React.Component {
         // Unless the play is actually free or a discard, decrease the amount
         // of available mana by the cost the card
         if (!(options.free || options.discard)) {
-          newState.mana -= state.deck[deckIds.indexOf(id)].mana
+          newState.mana -= card.mana
```

Comment:
The code searches twice for the card data - it is already stored in the `card` variable